### PR TITLE
prometheus: record IP AddressFamily

### PR DESCRIPTION
**Pull request by jzelinskie**

### CODE REVIEW

**Comment by mrd0ll4r** [comment]
I thought about this some more and don't think we need it for the frontends. Each frontend only listens on one address, which is either IPv4 or IPv6, so all traffic it gets will be either IPv4 or IPv6.

It could be useful for the storage, because you _could_ run HTTP on one address family and UDP on the other, however much sense that might make.

**Comment by jzelinskie** [comment]
>Each frontend only listens on one address

If you listen on 0.0.0.0, you bind to all interfaces.

File: bittorrent/bittorrent.go [hunk 1]
```diff
@@ -120,7 +120,8 @@ type AddressFamily uint8
 
 // AddressFamily constants.
 const (
-	IPv4 AddressFamily = iota
+	Unknown AddressFamily = iota
```
Comment (mrd0ll4r) [inline bittorrent/bittorrent.go]: Let's not add a third type to this boolean enum.

File: frontend/http/frontend.go [hunk 2]
```diff
@@ -30,19 +30,27 @@ var promResponseDurationMilliseconds = prometheus.NewHistogramVec(
...
 		Buckets: prometheus.ExponentialBuckets(9.375, 2, 10),
 	},
-	[]string{"action", "error"},
+	[]string{"action", "address_family", "error"},
 )
 
 // recordResponseDuration records the duration of time to respond to a Request
 // in milliseconds .
-func recordResponseDuration(action string, err error, duration time.Duration) {
+func recordResponseDuration(action string, af bittorrent.AddressFamily, err error, duration time.Duration) {
```
Comment (mrd0ll4r) [inline frontend/http/frontend.go]: You could make this a pointer to an `AddressFamily` and pass `nil` to indicate an unknown AF.

Comment (jzelinskie) [inline frontend/http/frontend.go]: The pointer's bigger than the af. I think we should use 0x0 to be unknown.

**Pull request merged**